### PR TITLE
Set up Cypress with Firebase stuff

### LINKED ISSUE #101: Get Cypress testing setup with Firebase
This involves continuing from the [cypress-setup](https://github.com/mn-pollinators/buzz-about/commits/cypress-setup) branch. The main thing here is setting up the integration with Firebase (https://github.com/prescottprue/cypress-firebase) so that we can talk to the emulator and have consistent test users for authentication.

### PR DESCRIPTION
**Pull request by floogulinc**
Closes #101

### CODE REVIEW

**Comment by floogulinc** [comment]
TODO: Document `serviceAccount.json`

**Comment by floogulinc** [comment]
There are docs for setting up the service account stuff here: https://github.com/mn-pollinators/buzz-about/blob/dev-docs/SETUP.md#setting-up-firebase-for-cypress

To build the app and run the Cypress end-to-end tests, start the Firebase emulators and run:
```
npm run e2e
```

To open Cypress (so you can run the tests interactively), start both the Firebase emulators and the development server and run:
```
npm run cy:open
```

**Review by helloworld12321 — CHANGES_REQUESTED** [review]
This looks wonderful—thank you, Paul! I've made a few minor notes (really tiny ones), but all in all, I'm really excited about this. Can't wait to start writing tests! :)

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Cypress Tests
```
[the remark this answers is not in the document]
Comment (floogulinc) [reply .github/workflows/cypress.yml:1]: I thought we might do that in a separate PR but I can do it here, the question is whether to use the badge for the GH action or the one from the Cypress dashboard.

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    - name: Cypress run
+      uses: cypress-io/github-action@v2
+      with:
+        record: true
+        start: npm run firebase:emulator
+        command: npm run e2e -- --record
+      env:
+        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+        SERVICE_ACCOUNT: ${{ secrets.FIREBASE_SERVICE_ACCOUNT }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (floogulinc) [reply .github/workflows/cypress.yml:30]: The Cypress GitHub action runs `npm install` so we need that for the post-install script.

File: .github/workflows/cypress.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (helloworld12321) [reply .github/workflows/cypress.yml:1]: Ooh, if it's convenient for you, I like the idea of having one from the Cypress dashboard! (That sounds fancy.)

File: .github/workflows/cypress.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (helloworld12321) [reply .github/workflows/cypress.yml:30]: Gotcha—neat, thanks!

**Review by helloworld12321 — APPROVED** [review]
Thumbs up! :+1: This looks like a really good tool.

**Review by kklamberty — APPROVED** [review]

**Pull request merged**